### DISCUSSION TITLE
Please prominently display the latest version of ipxe on the ipxe.org home page and/or download page

### DISCUSSION THREAD

**Discussion by shodanx2** [ipxe/ipxe#764] [General]
Despite the abundance of documentation available, many basic facts are missing from the website.
I decided to highlight this one which should be available at a glance.

I looked at the home page, the download page, the github release page (there are none) and at this point in time I still don't know what the latest version is. 

I have been experimenting with ipxe in my local network. I noticed in the ipxe boot banner that I had version 1.0.0 . But then in proxmox I noticed my clients were running ipxe 1.2.0 ! So, having now figured I was working with an outdated version, I need to find what is the latest version. And still, I cannot find this information !

**Comment by stappersg** [comment]
Consider to share with us where the version 1.0.0 came from.

**Reply by shodanx2** [reply]
It is from a ipxe build web ui from https://github.com/xbgmsharp/ipxe-buildweb
It's running in docker, in an lxc container in my proxmox machine 

Apparently it calls anything it builds "1.0.0" but my real version is "081b3" although that number is not visible on the banner of this executable

I found an open issue about this

https://github.com/xbgmsharp/ipxe-buildweb/issues/39

![image](https://user-images.githubusercontent.com/10621885/193570242-7138bf5d-cf39-4847-ab99-a03c1f0f2f69.png)


![image](https://user-images.githubusercontent.com/10621885/193569665-42bbc46d-4f6a-43dd-b5d5-0103c9dcb06f.png)

**Reply by stappersg** [reply]
> It is from a ipxe build web ui from https://github.com/xbgmsharp/ipxe-buildweb It's running in docker, in an lxc container in my proxmox machine

Somehow I do read that as _"I have a build system, it is even buzz word compliant!"_  And do read that as _I don't wanna judge a build system that I never had a close look at_.

> Apparently it calls anything it builds "1.0.0" but my real version is "081b3" although that number is not visible on the banner of this executable

```text
$ git show 081b3 | head -n 3
commit 081b3eefc48953795122c4b1a38430498e2bb2b6
Author: Michael Brown <mcb30@ipxe.org>
Date:   Mon Sep 19 17:49:25 2022 +0100
$
```
So the `081b3` does match latest of my git clone. Regarding the _Apparently it calls anything it builds "1.0.0"_ I'm thinking more _it does **not** use an iPXE git clone_. Yeah, read that as _I doubt the exotic build system_.  That https://github.com/xbgmsharp/ipxe-buildweb/issues/39 is being ignored for four years is a red flag.

**Comment by NiKiZe** [comment]
Latest version is git master, as you can find at https://github.com/ipxe/ipxe/tree/master (or just https://github.com/ipxe/ipxe)

From https://ipxe.org/download
> Source code
The iPXE source code is maintained in a git repository at https://github.com/ipxe/ipxe. You can check out a copy of the code using:
  git clone https://github.com/ipxe/ipxe.git
  
If iPXE was built properly you have the git commit id in the version string, check that against git history.

As a base reference you can also check git tags: https://github.com/ipxe/ipxe/tags

There is also #544 which somewhat covers this

**Comment by stappersg** [comment]
> > >  ... time wasted on 1.0 version ...

> > Consider to share with us where the version 1.0.0 came from.

> It is from a ipxe build web ui from https://github.com/xbgmsharp/ipxe-buildweb
> Apparently it calls anything it builds "1.0.0"

Regarding  the _"anything is 1.0.0"_: Could it be like an error I learnt lot from? I was in loop of
* same result
* edit
* build ( `make` )
* test again

Escape of that loop was realizing that I was testing each time the same binary.


Advice:
* build
* checksum the binary ( `sha1sum`, `md5sum`, _whatever checksum you like_ )
* build a known different binary
* checksum that binary
* compare the checksum
* know for sure that the binaries differ

Than make sure that you are realy using the iPXE binary that you want to use.

Request:  report back

**Reply by stappersg** [reply]
Reminder on the request of reporting back.

**Comment by shodanx2** [comment]
Ok, I've restarted the ipxe-buildweb container

![image](https://user-images.githubusercontent.com/10621885/195808896-2d56e3d7-cd1a-414e-a0ee-53bd6f2230fa.png)

Ran with this build config

http://ipxe-buildweb.lan/build.fcgi?BINARY=snponly.efi&BINDIR=bin-x86_64-efi&REVISION=master&DEBUG=&EMBED.00script.ipxe=%23%21ipxe%0Achain%20http%3A//router.lan/boot.ipxe&
Editable

This created a file called sponly.efi

it contains the script
chain http://router.lan/boot.ipxe

I placed this file at http://router.lan/snponly-ipxebuildweb.efi

I have a tftpserver, serving probably this file https://boot.ipxe.org/ipxe.efi from 2022-10-04 (can't match the hash with today's file)

![image](https://user-images.githubusercontent.com/10621885/195814889-d8ad24a5-151a-4f65-9cea-47ada7c265ac.png)


So, I fire up a EFI Q35 proxmox VM , this is what it looks like

https://user-images.githubusercontent.com/10621885/195812058-659e73e9-ea84-4704-b48a-8456fc10a8a7.mp4


So this snponly.efi file, shows version 1.0.0+ and not the build number in the banner


router.lan is a latest version openwrt router
it's running dnsmasq by default
here's the important part of the config file, it doesn't work entirely

dhcp-match=set:x86-legacy,option:client-arch,0
dhcp-match=set:x86-uefi,option:client-arch,6
dhcp-match=set:x86_64-uefi,option:client-arch,7
dhcp-match=set:x86_64-uefi,option:client-arch,9
#dhcp-match=set:arm32-uefi,option:client-arch,a
#dhcp-match=set:arm64-uefi,option:client-arch,b
dhcp-match=set:ipxe,77,"iPXE"

dhcp-boot=tag:x86-legacy,ipxe.x86.undionly.kpxe,router.lan,192.168.1.1
dhcp-boot=tag:x86-uefi,ipxe.x86.efi,router.lan,192.168.1.1
dhcp-boot=tag:x86_64-uefi,ipxe.x86_64.efi,router.lan,192.168.1.1
#dhcp-boot=tag:arm32-uefi,ipxe.arm32.efi,router.lan,192.168.1.1
#dhcp-boot=tag:arm64-uefi,ipxe.arm64.snponly.efi,router.lan,192.168.1.1
dhcp-boot=tag:ipxe,http://router.lan/boot.ipxe,router.lan,192.168.1.1

So, something is wrong in the ipxe-buildweb config that it does not update the banner for some reason

Also, it doesn't boot because it can't read from http://router.lan/boot.ipxe, but maybe it's a config problem

Still, all this to say, https://ipxe.org/download  should show the current versions somewhere in the first paragraph

It sould also have a link to the important files in boot.ipxe.org with a short explanation of what each is for.

As for my current setup, I used the files from https://ipxe.org/download and set then up with dhcp-boot=tag:ipxe,http://router.lan/boot.ipxe,router.lan,192.168.1.1    for now that works on EFI x86_64 and legacy bios x86_64 

here are the contents of boot.ipxe, it's a bit of a mess I'm trying stuff out (iscsi booting a windows installer to install on another iscsi boot drive)

```
#!ipxe

#console --x 1024 --y 768
#console --picture http://router.lan/ipxe.png

menu
item --gap -- ---------------- iPXE boot menu ----------------
item shell iPXE shell
item winpe winpe on samba.lan
item win20h2 win20h2 on samba.lan
item WinPE_amd64  Plain WinPE amd64
item iscsiwin iscsi windows installer
item iscsiwin2 iscsi windows installer
item iscsiwin3 iscsi windows installer
item ampe AMPE WinPE
item testsnp test ipxe-buildweb new snponly.efi file

item exit  Exit to BIOS
choose --default exit --timeout 10000 option && goto ${option}

:exit
exit
:shell
shell

:win20h2
chain http://samba.lan/win10-20h2/win20h2.ipxe

:winpe 
chain http://samba.lan/bartpe/winpe.ipxe

:ampe 
chain http://samba.lan/ampe/ampe.ipxe

:WinPE_amd64 
chain http://samba.lan/WinPE_amd64/WinPE_amd64.ipxe

:iscsiwin
sanhook iscsi:192.168.1.20:::1:iqn.2022-10.lan.proxmox:target01
sanboot iscsi:192.168.1.20:::6:iqn.2022-10.lan.proxmox:wininstall

:iscsiwin2
sanhook --drive 0x81 iscsi:192.168.1.20:::1:iqn.2022-10.lan.proxmox:target01 
sanboot --no-describe iscsi:192.168.1.20:::6:iqn.2022-10.lan.proxmox:wininstall

:iscsiwin3
sanhook --drive 0x80 iscsi:192.168.1.20:::1:iqn.2022-10.lan.proxmox:target01 
sanhook --drive 0x81 iscsi:192.168.1.20:::6:iqn.2022-10.lan.proxmox:wininstall
sanboot --no-describe --drive 0x81

:testsnp 
chain http://router.lan/snponly-ipxebuildweb.efi
```

**Reply by stappersg** [reply]
> So, something is wrong in the ipxe-buildweb config that it does not update the banner for some reason

Please change this _general discussion_ into a _Q&A discussion_ and mark it as **answered**.

Solve the _ipxe-buildweb-problem_ at https://github.com/xbgmsharp/ipxe-buildweb (not here at _iPXE_).

**Comment by shodanx2** [comment]
Did another try

in ipxebuildweb I changed from snponly.efi to "EFI PXE bootstrap 64-bit (.efi)"

http://ipxe-buildweb.lan/build.fcgi?BINARY=ipxe.efi&BINDIR=bin-x86_64-efi&REVISION=master&DEBUG=&EMBED.00script.ipxe=%23%21ipxe%0Achain%20http%3A//router.lan/boot.ipxe&

here is what booting that is like (same thing)

here are those two files
[ipxe bootloaders.zip](https://github.com/ipxe/ipxe/files/9784743/ipxe.bootloaders.zip)

This is what it looks booting it


https://user-images.githubusercontent.com/10621885/195818757-c1906849-6e9b-4616-9803-8862232ad311.mp4

**Comment by shodanx2** [comment]
The title of this discussion is "display the latest version of ipxe on the ipxe.org home page and/or download page"

Software should show it's current version on its download page. It's not really anything to do wit ipxebuildweb

But I will also tell them

**Reply by sebaxakerhtc** [reply]
> But I will also tell them

https://github.com/xbgmsharp/ipxe-buildweb/issues/65

**Reply by shodanx2** [reply]
Thanks
Makes it very confusing to know at a glance what version you have
But since then I've found you you don't really need to build it, just use 
https://boot.ipxe.org/ipxe.pxe

and this in your dhcp server and it works in most cases and it will show the version
```
dhcp-match=set:ipxe,77,"iPXE"
dhcp-boot=tag:ipxe,http://router.lan/boot.ipxe,router.lan,192.168.1.1
```

**Reply by sebaxakerhtc** [reply]
> Thanks Makes it very confusing to know at a glance what version you have But since then I've found you you don't really need to build it, just use https://boot.ipxe.org/ipxe.pxe
> 
> and this in your dhcp server and it works in most cases and it will show the version
> 
> ```
> dhcp-match=set:ipxe,77,"iPXE"
> dhcp-boot=tag:ipxe,http://router.lan/boot.ipxe,router.lan,192.168.1.1
> ```

I moved forward and create my own project [ipxe-simple](https://github.com/sebaxakerhtc/ipxe-simple)

**Reply by shodanx2** [reply]
cool

nice banner  !
![image](https://user-images.githubusercontent.com/10621885/197706388-4927e957-d3ee-4664-b4bb-4e996e28cb21.png)

and wifi support ? That's possible ?  I will try that at some point thanks !

**Reply by stappersg** [reply]
> and wifi support ? That's possible ? I will try that at some point thanks !

I'm looking forward to such feedback.

**Reply by sebaxakerhtc** [reply]
> cool
> 
> nice banner ! ![image](https://user-images.githubusercontent.com/10621885/197706388-4927e957-d3ee-4664-b4bb-4e996e28cb21.png)
> 
> and wifi support ? That's possible ? I will try that at some point thanks !

Yes, of cource. WiFi support have the latest versions with WEP, but there's a branch to support WPA2-PSK on older one.
Feel free to try it. The link to this branch is in readme

**Comment by tonho911** [comment]
> Despite the abundance of documentation available, many basic facts are missing from the website. I decided to highlight this one which should be available at a glance.
> 
> I looked at the home page, the download page, the github release page (there are none) and at this point in time I still don't know what the latest version is.
> 
> I have been experimenting with ipxe in my local network. I noticed in the ipxe boot banner that I had version 1.0.0 . But then in proxmox I noticed my clients were running ipxe 1.2.0 ! So, having now figured I was working with an outdated version, I need to find what is the latest version. And still, I cannot find this information !

How do I set up workflows in docker.yml, bro? I tried it and it doesn't work either. After creating a github docker secret, what else is missing?

**Comment by shodanx2** [comment]
So what is the current version of iPXE right now ?

https://ipxe.org/start
https://ipxe.org/download

These pages have no version numbers on them

What is the version of those files ?

```
https://boot.ipxe.org/ipxe.efi
https://boot.ipxe.org/ipxe.pxe
https://boot.ipxe.org/undionly.kpxe
https://boot.ipxe.org/snponly.efi
```

And the reason I'm asking is that  I have  a problem and maybe I just have a version that's too old..
So knowing the current version would be useful.

I have version 1.21

Wikipedia says that's the latest version
<img width="513" height="395" alt="image" src="https://github.com/user-attachments/assets/14b009d4-1c7a-406d-852c-f045ccb814e6" />

But usually software website shows the current version of the software ...

**Reply by NiKiZe** [reply]
> So what is the current version of iPXE right now ?
> 

https://github.com/ipxe/ipxe/commits/master/
You build latest version from git master. That way you can easily update and rebuild and be sure you have the latest current version.
Especially if you have a problem you will be needed to build from sources.

> https://ipxe.org/start https://ipxe.org/download
> 

Correct, since you see this in git HEAD, possibly in the future you will find releases on the GitHub page, not on the ipxe.org wiki.
